lib/index.js

```diff
@@ -24,19 +24,30 @@ class ElectronDownloader {
   }
 
   get baseUrl () {
-    return process.env.NPM_CONFIG_ELECTRON_MIRROR ||
+    return process.env.ELECTRON_MIRROR ||
+      process.env.npm_package_config_electron_mirror ||
       process.env.npm_config_electron_mirror ||
-      process.env.ELECTRON_MIRROR ||
+      process.env.NPM_CONFIG_ELECTRON_MIRROR ||
       this.opts.mirror ||
       'https://github.com/electron/electron/releases/download/v'
   }
 
   get middleUrl () {
-    return process.env.ELECTRON_CUSTOM_DIR || this.opts.customDir || this.version
+    return process.env.ELECTRON_CUSTOM_DIR ||
+      process.env.npm_config_electron_custom_dir ||
+      process.env.NPM_CONFIG_ELECTRON_CUSTOM_DIR ||
+      process.env.npm_package_config_electron_custom_dir ||
+      this.opts.customDir ||
+      this.version
   }
 
   get urlSuffix () {
-    return process.env.ELECTRON_CUSTOM_FILENAME || this.opts.customFilename || this.filename
+    return process.env.ELECTRON_CUSTOM_FILENAME ||
+      process.env.npm_config_electron_custom_filename ||
+      process.env.NPM_CONFIG_ELECTRON_CUSTOM_FILENAME ||
+      process.env.npm_package_config_electron_custom_filename ||
```

Comment:
Thinking about it, I can see why this may not be all that useful to set in a consuming project's package.json since the filename part is (the only?) platform specific bit.